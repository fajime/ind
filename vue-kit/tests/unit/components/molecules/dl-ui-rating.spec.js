import { mount } from '@vue/test-utils';
import { nextTick } from 'vue';

import DlUiRating from '../../../../src/components/molecules/dl-ui-rating';

describe('dl-ui-rating', () => {

  const wrapper = mount(DlUiRating);

  it('Component Name', () => expect(DlUiRating.name).toBe('dl-ui-rating'));

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('watch', () => {
    it('disable', async () => {
      const wrapperTest = mount(DlUiRating);
      wrapperTest.element.classList.add('fakeClass');
      await wrapperTest.setProps({ disabled : true });
      await nextTick();

      expect(wrapperTest.element.classList.length).toBe(4);
      expect(wrapperTest.element.classList[1]).toEqual('fakeClass');
      expect(wrapperTest.element.classList[3]).toEqual('fakeClass--disabled');
      await wrapperTest.setProps({ disabled : false });
      await nextTick();

      expect(wrapperTest.element.classList.length).toBe(2);
      expect(wrapperTest.element.classList[1]).toEqual('fakeClass');
    });

    it('readOnly', async () => {
      const wrapperTest = mount(DlUiRating);
      wrapperTest.element.classList.add('fakeClass');
      await wrapperTest.setProps({ readOnly : true });
      await nextTick();

      expect(wrapperTest.element.classList.length).toBe(4);
      expect(wrapperTest.element.classList[1]).toEqual('fakeClass');
      expect(wrapperTest.element.classList[3]).toEqual('fakeClass--readOnly');
      await wrapperTest.setProps({ readOnly : false });
      await nextTick();

      expect(wrapperTest.element.classList.length).toBe(2);
      expect(wrapperTest.element.classList[1]).toEqual('fakeClass');
    });
  });

  describe('methods', () => {
    describe('onStarClick', () => {
      it('readOnly : true, disabled : true', async () => {
        const stub = jest.fn();
        const wrapperTest = mount(DlUiRating);
        await wrapperTest.setProps({ readOnly : true, disabled : true });
        await nextTick();

        wrapperTest.vm.updateModel = stub;
        wrapperTest.vm.onStarClick(undefined, 1);
        expect(stub).not.toHaveBeenCalled();
      });
      it('readOnly : true, disabled : false', async () => {
        const stub = jest.fn();
        const wrapperTest = mount(DlUiRating);
        await wrapperTest.setProps({ readOnly : true, disabled : false });
        await nextTick();

        wrapperTest.vm.updateModel = stub;
        wrapperTest.vm.onStarClick(undefined, 2);
        expect(stub).not.toHaveBeenCalled();
      });
      it('readOnly : false, disabled : true', async () => {
        const stub = jest.fn();
        const wrapperTest = mount(DlUiRating);
        await wrapperTest.setProps({ readOnly : false, disabled : true });
        await nextTick();

        wrapperTest.vm.updateModel = stub;
        wrapperTest.vm.onStarClick(undefined, 3);
        expect(stub).not.toHaveBeenCalled();
      });
      it('readOnly : false, disabled : false', async () => {
        const stub = jest.fn();
        const wrapperTest = mount(DlUiRating);
        await wrapperTest.setProps({ readOnly : false, disabled : false });
        await nextTick();

        wrapperTest.vm.updateModel = stub;
        wrapperTest.vm.onStarClick(undefined, 4);
        expect(stub).toHaveBeenCalled();
      });
    });

    describe('onCancelClick', () => {
      it('readOnly : true, disabled : true', async () => {
        const stub = jest.fn();
        const wrapperTest = mount(DlUiRating);
        await wrapperTest.setProps({ readOnly : true, disabled : true });
        await nextTick();

        wrapperTest.vm.updateModel = stub;
        wrapperTest.vm.onCancelClick(undefined);
        expect(stub).not.toHaveBeenCalled();
      });
      it('readOnly : true, disabled : false', async () => {
        const stub = jest.fn();
        const wrapperTest = mount(DlUiRating);
        await wrapperTest.setProps({ readOnly : true, disabled : false });
        await nextTick();

        wrapperTest.vm.updateModel = stub;
        wrapperTest.vm.onCancelClick(undefined);
        expect(stub).not.toHaveBeenCalled();
      });
      it('readOnly : false, disabled : true', async () => {
        const stub = jest.fn();
        const wrapperTest = mount(DlUiRating);
        await wrapperTest.setProps({ readOnly : false, disabled : true });
        await nextTick();

        wrapperTest.vm.updateModel = stub;
        wrapperTest.vm.onCancelClick(undefined);
        expect(stub).not.toHaveBeenCalled();
      });
      it('readOnly : false, disabled : false', async () => {
        const stub = jest.fn();
        const wrapperTest = mount(DlUiRating);
        await wrapperTest.setProps({ readOnly : false, disabled : false });
        await nextTick();

        wrapperTest.vm.updateModel = stub;
        wrapperTest.vm.onCancelClick(undefined);
        expect(stub).toHaveBeenCalled();
      });
    });

    it('updateModel', () => {
      const wrapperTest = mount(DlUiRating);
      wrapperTest.vm.updateModel();
      expect(wrapperTest.emitted('update:modelValue')).toBeTruthy();
    });
  });
});
