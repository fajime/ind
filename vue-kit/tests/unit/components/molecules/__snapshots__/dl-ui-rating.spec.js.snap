// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dl-ui-rating renders correctly 1`] = `
<div
  class="dl-ui-rating"
>
  <span
    class="dl-ui-rating__cancel"
  >
    <div
      class="dl-ui-icon dl-ui-rating__icon"
    >
      <svg>
        <!--v-if-->
        <!--v-if-->
        <use
          xlink:href="#dl-ids-icon-void-outlined"
        />
      </svg>
    </div>
  </span>
  
  <span
    class="dl-ui-rating__star"
  >
    <div
      class="dl-ui-icon dl-ui-rating__icon"
    >
      <svg>
        <!--v-if-->
        <!--v-if-->
        <use
          xlink:href="#dl-ids-icon-star"
        />
      </svg>
    </div>
  </span>
  <span
    class="dl-ui-rating__star"
  >
    <div
      class="dl-ui-icon dl-ui-rating__icon"
    >
      <svg>
        <!--v-if-->
        <!--v-if-->
        <use
          xlink:href="#dl-ids-icon-star"
        />
      </svg>
    </div>
  </span>
  <span
    class="dl-ui-rating__star"
  >
    <div
      class="dl-ui-icon dl-ui-rating__icon"
    >
      <svg>
        <!--v-if-->
        <!--v-if-->
        <use
          xlink:href="#dl-ids-icon-star"
        />
      </svg>
    </div>
  </span>
  <span
    class="dl-ui-rating__star"
  >
    <div
      class="dl-ui-icon dl-ui-rating__icon"
    >
      <svg>
        <!--v-if-->
        <!--v-if-->
        <use
          xlink:href="#dl-ids-icon-star"
        />
      </svg>
    </div>
  </span>
  <span
    class="dl-ui-rating__star"
  >
    <div
      class="dl-ui-icon dl-ui-rating__icon"
    >
      <svg>
        <!--v-if-->
        <!--v-if-->
        <use
          xlink:href="#dl-ids-icon-star"
        />
      </svg>
    </div>
  </span>
  
</div>
`;
